federation/entities.py

```diff
@@ -0,0 +1,40 @@
+from abc import (
+    abstractmethod,
+    ABC,
+)
+from typing import (
+    List,
+    Any,
+)
+
+from hiku.graph import (
+    Field,
+)
+
+
+class FederatedResolver(ABC):
+    @abstractmethod
+    def resolve_references(
+        self,
+        refs: List[dict],
+        fields: List[Field]
+    ) -> List[Any]:
+        pass
+
+
+def resolve_entities(entities_link, items):
+    items_map = {item.name: item for item in items}
+
+    _entity_fields = entities_link.node.fields
+    representations = entities_link.options.get('representations')
+
+    entity_type = representations[0]['__typename']
```

Comment:
Apollo Gateway splits representations by __typename, so it is kind of safe to handle just one resolver.
But theoretically, there can be several distinct __typename in representations (I've done it by querying _entities directly on the service graph).

Some implementations in kotlin and graphene for example:
- https://github.com/ExpediaGroup/graphql-kotlin/blob/master/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/execution/EntityResolver.kt
- https://github.com/preply/graphene-federation/blob/master/graphene_federation/entity.py